src/main/java/seedu/sudohr/model/SudoHr.java

```diff
@@ -178,6 +194,106 @@ public void removeEmployeeFromDepartment(Person p, Department d) {
 
     //// event-level operations
 
+    /**
+     * Adds a event to the address book.
+     * The event must not already exist in the address book.
+     */
+    public void addEvent(Event event) {
+        events.addEvent(event);
+    }
+
+    /**
+     * Removes {@code event} from this {@code SudoHr}.
+     * {@code event} must exist in the sudohr book.
+     */
+    public void deleteEvent(Event event) {
+        events.remove(event);
+    }
+
+    /**
+     * Replaces the given event {@code eventToEdit} in the list with
+     * {@code editedEvent}.
+     * {@code eventToEdit} must exist in the sudohr book.
+     * The person identity of {@code editedEvent} must not be the same as another
+     * existing person in the sudohr book.
+     */
+    public void setEvent(Event eventToEdit, Event editedEvent) {
+        requireNonNull(editedEvent);
+        events.setEvent(eventToEdit, editedEvent);
+    }
+
+    /**
+     * Returns true if a person with the same identity as {@code event} exists in
+     * the sudohr book.
+     */
+    public boolean hasEvent(Event event) {
+        requireNonNull(event);
+        return events.contains(event);
+    }
+
+    /**
+     * Returns true if a person with the same identity as {@code person} exists in the specified event
+     * {@code event}
+     * the sudohr book.
+     */
+    public boolean hasEmployeeInEvent(Event event, Person person) {
+        requireAllNonNull(event, person);
+        return event.hasPerson(person);
+    }
+
+    /**
+     * Adds a person {@code person} to a specified event{@code event}
+     * in the sudohr book.
+     */
+    public void addEmployeeToEvent(Event event, Person person) {
+        requireAllNonNull(event, person);
+        event.addPerson(person);
+    }
+
+    /**
+     * Deletes a person {@code person} from a specified event{@code event}
+     * in the sudohr book.
+     */
+    public void deleteEmployeeFromEvent(Event event, Person person) {
+        requireAllNonNull(event, person);
+        event.deletePerson(person);
+    }
+
+    /**
+     * Update a person {@code person} with editedPerson {@code person} in all events
+     * in the sudohr book.
+     */
+    public void cascadeUpdateUserInEvents(Person personToEdit, Person editedPerson) {
```

Comment:
Same concern as before, what if the edited person changed its phone number (or email) which clashes with someone already added to the event? Will there be checks in place?

By right, your part shouldn't need to concern with this because when the user edits a person to have duplicated email/phone fields, there should be an error thrown before your part of editing a user in an event list kicks in. Worried there might be integration issues but if the flow is as above, should be fine right..? Or should we have defensive checks also lol